crates/tests/src/e2e/ledger_tests.rs

```diff
@@ -2808,3 +2820,407 @@ fn test_genesis_manipulation() -> Result<()> {
+        ibc_tests::create_channel_with_hermes(&test, &test_gaia)?;
+
+    // Start relaying
+    let hermes = ibc_tests::run_hermes(&test)?;
+    let _bg_hermes = hermes.background();
+
+    let validator_one_rpc = get_actor_rpc(&test, Who::Validator(0));
+
+    let mut proposal_id: u64 = 0;
+    let mut submit_proposal = |tx: TestWasms| -> Result<Epoch> {
+        // Use minimum offsets for faster proposals
+        const VOTING_END_OFFSET: u64 = 1;
+        const ACTIVATION_OFFSET: u64 = 1;
+
+        let next_epoch = get_epoch(&test, &validator_one_rpc)?.next();
+        let activation_epoch =
+            next_epoch + VOTING_END_OFFSET + ACTIVATION_OFFSET;
+
+        let valid_proposal_json_path = prepare_proposal_data(
+            test.test_dir.path(),
+            defaults::albert_address(),
+            tx.read_bytes(),
+            next_epoch.0,
+            Some(VOTING_END_OFFSET),
+            Some(ACTIVATION_OFFSET),
+        );
+
+        let submit_proposal_args = apply_use_device(vec![
+            "init-proposal",
+            "--data-path",
+            valid_proposal_json_path.to_str().unwrap(),
+            "--gas-limit",
+            "2200000",
+            "--node",
+            &validator_one_rpc,
+        ]);
+        let mut client =
+            run!(test, Bin::Client, submit_proposal_args, Some(40))?;
+        client.exp_string(TX_APPLIED_SUCCESS)?;
+        client.assert_success();
+
+        // Start the voting epoch
+        let _epoch = epoch_sleep(&test, &validator_one_rpc, 120)?;
+
+        let proposal_id_str = proposal_id.to_string();
+        let submit_proposal_vote = apply_use_device(vec![
+            "vote-proposal",
+            "--proposal-id",
+            &proposal_id_str,
+            "--vote",
+            "yay",
+            "--address",
+            "validator-0",
+            "--node",
+            &validator_one_rpc,
+        ]);
+
+        let mut client =
+            run!(test, Bin::Client, submit_proposal_vote, Some(40))?;
+        client.exp_string(TX_APPLIED_SUCCESS)?;
+        client.assert_success();
+
+        proposal_id += 1;
+
+        Ok(activation_epoch)
+    };
+
+    // Propose phase 2 - staking party 🥳
+    let activation_epoch = submit_proposal(TestWasms::TxProposalPhase2)?;
+
+    // Wait for phase 2 proposal activation
+    let mut current_epoch = get_epoch(&test, &validator_one_rpc)?;
+    while current_epoch < activation_epoch {
+        // There should be no staking rewards before phase 2
+        let staking_rewards =
+            query_staking_rewards(&test, "validator-0", &validator_one_rpc)?;
+        assert_eq!(staking_rewards, token::Amount::zero());
+
+        // There should be no tokens in PGF address
+        let balance = find_balance(
+            &test,
+            Who::Validator(0),
+            NAM,
+            PGF_ADDRESS,
+            None,
+            None,
+        )?;
+        assert_eq!(balance, token::Amount::zero());
```

Comment:
can we also test other things in Phase 1 just for completeness? Such as:
- cannot transfer any NAM
- cannot IBC transfer
- can bond, unbond, and withdraw

Reply:
yup totally

Reply:
added these too